src/components/AttachmentCarousel/index.js

```diff
@@ -177,6 +177,9 @@ class AttachmentCarousel extends React.Component {
             throw new Error('Attachment not found');
         }
 
+        // Update the parent modal's state with the source and name from the mapped attachments
+        this.props.onNavigate(attachments[page]);
```

Comment:
This adds the fileName and `isAuthTokenRequired` to the initial `source`  in the parent component
We already use it to add the same metadata while attachments are changed using the carousel 

I would prefer if this logic can be moved outside. `isAuthTokenRequired` is straightforward - attachments from the path `/chat-attachments/...` require auth, others don't. But the original file name can only be extracted out of the raw html comment

This attachment metadata extracting logic can be moved to be at the same level, where we parse the report action JSON data saved in Onyx